Source/Scene/Model.js

```diff
@@ -693,22 +748,42 @@ define([
             setCachedGltf(model, cachedGltf);
             gltfCache[cacheKey] = cachedGltf;
 
-            loadText(url, options.headers).then(function(data) {
-                cachedGltf.gltf = JSON.parse(data);
-                var models = cachedGltf.modelsToLoad;
-                var length = models.length;
-                for (var i = 0; i < length; ++i) {
-                    var m = models[i];
-                    if (!m.isDestroyed()) {
-                        setCachedGltf(m, cachedGltf);
+            if (url.toLowerCase().indexOf('.bgltf', url.length - 6) !== -1) {
```

Comment:
It's better to use a regex for this `/\.bgltf$/i.test(url)` which returns true if the strings ends with `.bgltf`